components/Critical/SideProgressBar.tsx

```diff
@@ -102,12 +124,22 @@ function SideProgressBar() {
       )
     } else {
       return (
-        <ListItem key={text} onClick={() => handleListItemClick(index)}>
+        <ListItem
+          key={text}
+          onClick={async () => {
+            if (onNavigate) {
+              await onNavigate()
+            }
```

Comment:
this async function seems to be the same one as lines 104-108 -- worth abstracting?

Reply:
yeah